addons/product/models/product_pricelist.py

```diff
@@ -592,18 +594,25 @@ def _compute_price(self, price, price_uom, product, quantity=1.0, partner=False)
             # complete formula
             price_limit = price
             price = (price - (price * (self.price_discount / 100))) or 0.0
+            product_cur = product.cost_currency_id if self.base == 'standard_price' else product.currency_id
             if self.price_round:
                 price = tools.float_round(price, precision_rounding=self.price_round)
 
+            def _conv_curr_pricelist(price):
+                return self.currency_id._convert(price, product_cur, self.env.company, date, round=False) if price else price
```

Comment:
```suggestion
            def convert_to_base_price_currency(amount):
                return self.currency_id._convert(amount, price_currency, self.env.company, date, round=False) if amount else amount
```

* Clearer method name
* avoid confusing price variable name, even if it works fine, the variables scope is clearer with different names.